chain/client/src/chunks_delay_tracker.rs

```diff
@@ -58,17 +58,30 @@ impl ChunksDelayTracker {
                 }
             }
         }
-        near_metrics::set_gauge(&metrics::CHUNKS_RECEIVING_DELAY_US, max_delay as i64);
+        max_delay
+    }
+
+    fn update_chunks_metric(&mut self, head_height: BlockHeight) {
+        near_metrics::set_gauge(
+            &metrics::CHUNKS_RECEIVING_DELAY_US,
+            self.get_max_delay_us(head_height) as i64,
+        );
     }
 
     // Computes the difference between the latest block we are aware of and the current head.
-    fn update_blocks_ahead_metric(&mut self, head_height: BlockHeight) {
-        let value = if let Some((&latest, _)) = self.heights.iter().next_back() {
+    fn get_blocks_ahead(&mut self, head_height: BlockHeight) -> u64 {
+        if let Some((&latest, _)) = self.heights.iter().next_back() {
             latest.saturating_sub(head_height)
         } else {
             0
-        };
-        near_metrics::set_gauge(&metrics::BLOCKS_AHEAD_OF_HEAD, value as i64);
+        }
+    }
+
+    fn update_blocks_ahead_metric(&mut self, head_height: BlockHeight) {
+        near_metrics::set_gauge(
```

Comment:
This is not related to this PR. Probably we should refactor the code to have ability to check whenever metrics are updater properly.

Reply:
I don't see how to make it cleanly.
Either:
* I need a test-only API, which is prone to testing not production code but test-only code.
* Complicate the interface by adding some `Option<>` parameter which in prod is always `None`. This exposes implementation details to the interface.
* Maybe something in-between: let us register a callback in `Tracker` object. Then we can set expectations on the callback. However that needs a few more structs defined.

Reply:
Yes, if we want to do this, this will require creating `Metrics` structs and passing it down. So either `Tracker` will have to take it in constructor, or in the method that use metrics.

I'm not saying it's worth doing, but if we really want to test those metrics in unit tests, that's the only way I see right now,